packages/typespec-test/test/batch_modular/generated/typespec-ts/src/api/pagingHelpers.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  getPagedAsyncIterator,
+  PagedAsyncIterableIterator as CorePagedAsyncIterableIterator,
+  PagedResult,
+  PageSettings as CorePageSettings,
+} from "@azure/core-paging";
+import {
+  Client,
+  createRestError,
+  PathUncheckedResponse,
+} from "@azure-rest/core-client";
+import {
+  PageSettings,
+  PagedAsyncIterableIterator,
+} from "../models/pagingTypes.js";
+
+export function buildPagedAsyncIterator<
+  TElement,
+  TResponse extends PathUncheckedResponse = PathUncheckedResponse
+>(
+  client: Client,
+  getInitialResponse: () => PromiseLike<TResponse>,
+  processResponseBody: (result: TResponse) => Promise<unknown>
+): PagedAsyncIterableIterator<TElement> {
+  let firstRun = true;
+  let itemName: string, nextLinkName: string | undefined;
+  const firstPageLinkPlaceholder = "";
+  const pagedResult: PagedResult<TElement[]> = {
+    firstPageLink: firstPageLinkPlaceholder,
+    getPage: async (pageLink: string) => {
```

Comment:
It will be coverred in another issue https://github.com/Azure/autorest.typescript/issues/2134. Currently I am not clear if we could define them in typespec.

But for us it could be easy to implement by adding optional config to specify the calling.